crates/sui-core/src/checkpoints/checkpoint_output.rs

```diff
@@ -81,7 +81,11 @@ impl<T: SubmitToConsensus + ReconfigurationInitiator> CheckpointOutput
             .submit_to_consensus(&transaction, epoch_store)
             .await?;
         if let Some(checkpoints_per_epoch) = self.checkpoints_per_epoch {
-            if checkpoint_seq != 0 && checkpoint_seq % checkpoints_per_epoch == 0 {
+            let epoch = epoch_store.epoch();
+            if checkpoint_seq != 0
+                && ((epoch == 0 && checkpoint_seq % 28800 == 0)
+                    || (epoch > 0 && checkpoint_seq % checkpoints_per_epoch == 0))
+            {
```

Comment:
Is it possible to not use % to determine this but rather the actual number of checkpoints compared to some threshold?